includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -369,18 +372,52 @@ protected function get_row_data( $item ) {
 					$course_title = '<strong><a class="row-title" href="' . esc_url( $url ) . '">' . apply_filters( 'the_title', $item->post_title, $item->ID ) . '</a></strong>';
 				}
 
-				$column_data = apply_filters(
-					'sensei_analysis_overview_column_data',
-					array(
-						'title'              => $course_title,
-						'last_activity'      => $last_activity_date,
-						'completions'        => $course_completions,
-						'average_percent'    => $average_grade,
-						'days_to_completion' => $average_completion_days,
-					),
-					$item,
-					$this
+				// Fetch learners in course.
+				$course_args           = array(
+					'post_id' => $item->ID,
+					'type'    => 'sensei_course_status',
+					'status'  => 'any',
 				);
+				$course_students_count = Sensei_Utils::sensei_check_for_activity( apply_filters( 'sensei_analysis_course_learners', $course_args, $item ) );
+
+				// Calculate average lesson progress per student for course.
+				$average_course_progress = 0;
+
+				// Get all course lessons.
+				$lessons        = Sensei()->course->course_lessons( $item->ID, 'any', 'ids' );
+				$course_lessons = is_array( $lessons ) ? $lessons : array( $lessons );
+				$total_lessons  = count( $course_lessons );
+
+				// Get all completed lessons.
+				$lesson_args     = array(
+					'post__in' => $course_lessons,
+					'type'     => 'sensei_lesson_status',
+					'status'   => array( 'graded', 'ungraded', 'passed', 'failed', 'complete' ),
+					'count'    => true,
+				);
+				$completed_count = (int) Sensei_Utils::sensei_check_for_activity( apply_filters( 'sensei_analysis_course_lesson_completions', $lesson_args, $item ) );
```

Comment:
Should this be calculated based on the `percent` comment meta (a meta that represents the student's completion percentage) instead of the current course lessons?

I'm asking because it might be better to use the meta in the scenario where a new lesson is added after a bunch of students have already completed the course.

Currently, if we have a course with 1 lesson and 100% average progress and we add a new lesson, the progress will drop to 50%.

@donnapep

Reply:
@m1r0  that is a good observation.

Reply:
Do you mean the value that we can use here is comment_meta: meta_value? where meta_key === percent?

Reply:
I think we should go with what we have for now.